src/docs/sdk/performance/dynamic-sampling-context.mdx

```diff
@@ -0,0 +1,168 @@
+No information can be changed, added or deleted after the first propagation.
+
+### SDKs
+
+SDKs are responsible for propagating **Dynamic Sampling Context** across all applications that are part of a trace.
+This involves:
+
+1. Collecting the information that makes up the DSC **xor** extracting the DSC from incoming requests.
+2. Propagating DSC to downstream SDKs.
+3. Sending the DSC to Sentry via the `trace` envelope header.
+
+Because there are quite a few things to keep in mind for DSC propagation and to avoid every SDK running into the same problems, we defined a [unified propagation mechanism](#unified-propagation-mechanism) (step-by-step instructions) that all SDK implementations should be able to follow.
+
+## Baggage
+
+We chose `baggage` as the propagation mechanism for DSC. ([w3c baggage spec](https://www.w3.org/TR/baggage/))
+Baggage is a standard HTTP header with URI encoded key-value pairs.
+
+For the propagation of DSC, SDKs first read the DSC from the baggage header of incoming requests/messages.
+To propagate DSC to downstream SDKs/services, we create a baggage header (or modify an existing one) through HTTP request instrumentation.
+
+<Alert level="info">
+
+Other vendors might also be using the `baggage` header.
+If a `baggage` header already exists on an outgoing request, SDKs should aim to be good citizens by only **appending** Sentry values to the header.
+In the case that another vendor added Sentry values to an outgoing request, SDKs may overwrite those values.
+
+SDKs must not add other vendors' baggage from incoming requests to outgoing requests.
+Sentry SDKs only concern themselves with Sentry baggage.
+
+</Alert>
+
+The following is an example of what a baggage header containing Dynamic Sampling Context may look like:
+
+```
+baggage: other-vendor-value-1=foo;bar;baz, sentry-traceid=771a43a4192642f0b136d5159a501700, sentry-publickey=49d0f7386ad645858ae85020e393bef3; sentry-userid=Am%C3%A9lie, other-vendor-value-2=foo;bar;
+```
+
+See the [Payloads section](#payloads) for a complete list of key-value pairs that SDKs should propagate.
+
+## Payloads
+
+Dynamic Sampling Context is propagated via a baggage header and sent to Sentry via the `trace` envelope header.
+
+### Baggage-Header
+
+SDKs may set the following key-value pairs on baggage headers.
+While none of these values are strictly required, SDKs should make their best effort to add as many of them to the baggage header as possible when starting a trace.
+
+Some values are marked as "required for Dynamic sampling to work".
+Nothing bad is going to happen when these values are missing, however, Dynamic Sampling will not work without them.
+
+- `sentry-traceid` - **Required for Dynamic Sampling to work** - The original trace ID as generated by the SDK.
+- `sentry-publickey` - **Required for Dynamic Sampling to work** - Public key as defined by the user via the DSN in the SDK options.
```

Comment:
I would name this just `key`. We already no longer have a private key on DSNs in use any more and it shaves off some bytes.

Reply:
It's named `public_key` in Relay, it would be nice to make it just `key`, but I think the SDKs should follow the relay names for the dynamic sampling context as much as possible https://github.com/getsentry/relay/blob/b23b8d3b2360a54aaa4d19ecae0231201f31df5e/relay-sampling/src/lib.rs#L691.

Reply:
We should mention here that the plan for the future is to 1:1-dump key-value pairs from the `baggage` header directly onto the trace envelope header. If we are done with that, shorter/longer keys will not have an effect on bundle size anymore.